oio/cli/containersharding/sharding.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (C) 2020 OpenIO SAS, as part of OpenIO SDS
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 3.0 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library.
+
+
+from logging import getLogger
+
+from oio.cli import Lister
+from oio.common.json import json
+from oio.container.sharding import ContainerSharding
+
+
+class ContainerShardingCommandMixin(object):
+    """Command taking a root container as parameter"""
+
+    def patch_parser_container_sharding(self, parser):
+        parser.add_argument(
+            'root_container',
+            metavar='<root container>',
+            help=("Name of the root container to interact with.\n")
+        )
+
+
+class ReplaceContainerSharding(ContainerShardingCommandMixin, Lister):
+    """Replace current shards with the new configuration."""
+
+    log = getLogger(__name__ + '.ReplaceContainerSharding')
+
+    def get_parser(self, prog_name):
+        parser = super(ReplaceContainerSharding, self).get_parser(prog_name)
+        self.patch_parser_container_sharding(parser)
+        parser.add_argument(
+            'configuration',
+            metavar='<configuration>',
+            help="""
+                Sharding configuration
+                JSON Syntax:
+                [{"index": 0, "lower": "", "upper": "sharding"},
+                 {"index": 1, "lower": "sharding", "upper": ""}]
```

Comment:
Since `upper` of Nth shard must be the `lower` of Nth+1 shard,
we may ask to user to provide
```
{"index": "0", "upper": "cde"},
{"index": "1, "upper": "ghi"},
{"index": "2", "upper": "opq"},
{"index": "3:", "upper": ""},
```
and build the full JSON internally, 
It will simplify various checks in ContainerSharding

Reply:
Unless the user wishes, this JSON will be created by `openio container-sharding find` (not yet implemented), as for `swift`.
In addition, having the `lower` and `upper` ensures that there has been no input error.